addons/project/models/project_milestone.py

```diff
@@ -59,19 +64,20 @@ def _compute_task_count(self):
     def _compute_can_be_marked_as_done(self):
         if not any(self._ids):
             for milestone in self:
-                milestone.can_be_marked_as_done = not milestone.is_reached and all(milestone.task_ids.is_closed)
+                milestone.can_be_marked_as_done = not milestone.is_reached and all([True if task.state in CLOSED_STATES.keys() else False for task in milestone.task_ids])
```

Comment:
```suggestion
                milestone.can_be_marked_as_done = not milestone.is_reached and all(milestone.task_ids.mapped(lambda t: t.state in CLOSED_STATES))
```
- `True if expression else False` <=> `expression`
- Let's use `mapped`, as the framework provides it :slightly_smiling_face:
- `in dict.keys()` <=> `in dict`